filepicker/src/main/java/dev/arkbuilders/components/filepicker/ArkRootPickerFragment.kt

```diff
@@ -0,0 +1,97 @@
+package dev.arkbuilders.components.filepicker
+
+import androidx.core.os.bundleOf
+import androidx.fragment.app.setFragmentResult
+import androidx.lifecycle.lifecycleScope
+import dev.arkbuilders.arklib.data.folders.FoldersRepo
+import dev.arkbuilders.components.utils.hasNestedRoot
+import dev.arkbuilders.components.utils.isInternalStorage
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.nio.file.Path
+
+class ArkRootPickerFragment : ArkFilePickerFragment() {
+    private var rootNotFavorite = false
+
+    override fun onFolderChanged(currentFolder: Path) {
+        lifecycleScope.launch {
+            val folders = FoldersRepo.instance.provideFolders()
+            val roots = folders.keys
+
+            if (currentFolder.isInternalStorage() || currentFolder.hasNestedRoot(roots)) {
```

Comment:
Are we still allowing picking a root folder even if the folder is inside another root folder already?

Reply:
If folder has parental root, we should still be able to make it favorite